### PR TITLE
:wrench: Onboard a user

### PR DESCRIPTION
**Pull request by zwaans**
# Changes
introduced onboard endpoint

## Todo: 
- [x] What do we protect this with?

### CODE REVIEW

File: specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                  },
+                  "phone_number": {
+                    "type": "string",
+                    "example": "+44 20 7946 0958"
+                  },
+                  "shop_name": {
+                    "type": "string",
+                    "example": "Holmes Investigations"
+                  },
+                  "address": {
```
Comment (zwaans) [inline specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json:75]: How can we make certain fields in the `address` required?

File: specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    "$ref": "#/components/schemas/BaseAddress"
+                  },
```
Comment (M4tini) [inline specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json:77]: You can make `address` fields required by defining a second object it needs to adhere to.
Same goes for the `return_address` fields.
```suggestion
                  "address": {
                    "allOf": [
                      {
                        "$ref": "#/components/schemas/BaseAddress"
                      },
                      {
                        "required": [
                          "street_1",
                          "city",
                          "country_code"
                        ]
                      },
                    ]
                  },
```

**Review by M4tini — CHANGES_REQUESTED** [review]

File: specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json [hunk 3]
```diff
@@ -0,0 +1,119 @@
+{
+  "parameters": [
+    {
+      "$ref": "#/components/parameters/path-invitation_id"
+    }
+  ],
+  "post": {
+    "tags": [
+      "RPC"
```
Comment (M4tini) [inline specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json:9]:
```suggestion
      "Invitations"
```

File: specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    ],
+    "security": [
+      {
+        "OAuth2": [
+          "brokers.manage"
+        ]
+      }
+    ],
+    "summary": "Start onboarding process.",
+    "description": "This endpoint creates a user based on an invitation.",
```
Comment (M4tini) [inline specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json:19]:
```suggestion
    "description": "This `RPC` endpoint creates a user based on an invitation.",
```

File: specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    "requestBody": {
+      "description": "The onboarding request.",
+      "required": true,
+      "content": {
+        "application/vnd.api+json": {
```
Comment (M4tini) [inline specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json:24]:
```suggestion
        "application/json": {
```

**Review by M4tini — CHANGES_REQUESTED** [review]

**Review by M4tini — APPROVED** [review]

**Pull request merged**